### ISSUE TITLE
Community package requirements: sanity tests and repository management

### ISSUE THREAD

**Issue by gotmax23** [CheckPointSW/CheckPointAnsibleMgmtCollection#142]
(Note: This issue was filed in a semi-automated fashion on behalf of the Ansible Community Steering Committee. Let me know if you see errors in this issue.)

As per the [Ansible community package inclusion requirements][ci-testing], collections must pass `ansible-test sanity` tests. Version `5.2.3` of `check_point.mgmt`, corresponding to the `v5.2.3` tag in this repo, fails one or more of the required sanity tests.

The contents in the `v5.2.3` git tag do not match `check_point-mgmt-5.2.3.tar.gz` as uploaded to Ansible Galaxy. For future releases, please make sure that the contents uploaded to Galaxy match the sources that were tagged as that release. See the [Repository management requirements][repo-mgmt] for more information.

Please see the errors below and address them. If these issues aren't addressed within a reasonable time period, the collection may be subject to [removal from Ansible][removal].

Thank you for your efforts and for being part of the Ansible package! We appreciate it.

---

## Sanity tests

The following tests were run using `ansible-test` version `2.17.3`:

- ansible-doc
- compile
- validate-modules
- yamllint

Note that this is only a subset of the required sanity tests. Please make sure you run them in all in your CI.

### Results

> **💡 NOTE:**
>
> Check the `[explain]` links below for more information about each test and how to fix failures.
> See [Sanity Tests: Ignores](https://docs.ansible.com/ansible/latest/dev_guide/testing/sanity/ignores.html) in the dev guide if, after reading the test-specific documentation, you still believe an error is a false positive.

The test `ansible-test sanity --test validate-modules` [[explain](https://docs.ansible.com/ansible-core/2.17/dev_guide/testing/sanity/validate-modules.html)] failed with 261 errors:

**[Issue truncated. See full results at <https://github.com/ansible-community/package-test-results/blob/main/rendered/10.3.0/check_point.mgmt.md#results>]**

## File divergences

The following files differ between the `v5.2.3` git tag and `check_point-mgmt-5.2.3.tar.gz` on Ansible Galaxy:

- `plugins/module_utils/checkpoint.py` (`WRONG_HASH`)
- `changelogs/.plugin-cache.yaml` (`WRONG_HASH`)
- `changelogs/changelog.yaml` (`WRONG_HASH`)
- `CHANGELOG.rst` (`WRONG_HASH`)

[ci-testing]: https://docs.ansible.com/ansible/latest/community/collection_contributors/collection_requirements.html#ci-testing
[repo-mgmt]: https://docs.ansible.com/ansible/latest/community/collection_contributors/collection_requirements.html#repository-management
[removal]: https://github.com/ansible-collections/overview/blob/main/removal_from_ansible.rst

**Comment by gotmax23** [comment]
This is a follow up to https://github.com/CheckPointSW/CheckPointAnsibleMgmtCollection/issues/121, run against the latest version of ansible-core. Thank you for fixing the first issue so promptly! It looks like most of the issues here are caused by https://github.com/ansible/ansible/commit/92d2c66db2eecd0f5e26e6f49589dbcdb4612618#diff-2a423d6bc335536285d838786c00ea33a1361e6f5bde5746aa01d79a8d7cea34L179-L186 which removed `CertificateError` from the `ansible.modules_utils.urls` which this collection incidentally relied on.

Thanks again for being part of the Ansible package.

**Comment by chkp-edenbr** [comment]
Hi,
I wasn't able to replicate the issue on our latest release v6.1.1, seems like the files have the same md5 and the collection passed sanity tests.

**Comment by gotmax23** [comment]
Did you run the tests with the same version of ansible-core (2.17.3) that was used above? Please also make sure that you are running the required CI tests in this repository as explained in https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html#ci-testing.

**Comment by chkp-edenbr** [comment]
Yes, and there aren't 261 errors as mentioned when running ansible's sanity tests

**Comment by gotmax23** [comment]
How are you running the tests? What's the value of `ansible-test --version`?

**Comment by chkp-edenbr** [comment]
Re you running the test on the latest version and receiving those errors mentioned?

**Comment by gotmax23** [comment]
The sanity test issues are fixed as of 6.2.1 (see [the latest report for this collection](https://github.com/ansible-community/package-test-results/blob/main/rendered/11.0.0/check_point.mgmt.md)). Note the repository management issue in the report, however. For future releases, please make sure that the contents uploaded to Galaxy match the sources that were tagged as that release. Thank you!

**Comment by gotmax23** [comment]
Actually, I'll reopen this to keep track of the repository management issue. Please fix the file divergences identified [in the report](https://github.com/ansible-community/package-test-results/blob/main/rendered/11.0.0/check_point.mgmt.md) and then create a new Galaxy release so the fixes are caught in the next round of automated testing.
